lib-es5/api_client/execute_request.js

```diff
@@ -101,9 +103,10 @@ function execute_request(method, params, auth, api_url, callback) {
             query_params
           }
         };
-        deferred.reject(err_obj.error);
-        if (typeof callback === "function") {
+        if (didUserPassCB) {
           callback(err_obj);
+        } else {
+          deferred.reject(err_obj.error);
```

Comment:
We agreed to move all of those repeating 'if else' blocks with a 'handleError' method